OTAnalytics/domain/track.py

```diff
@@ -1,33 +1,100 @@
-from dataclasses import dataclass
 from datetime import datetime
 from pathlib import Path
 from typing import Iterable, Optional
 
+from pydantic import BaseModel, Field, validator
 
-@dataclass(frozen=True)
-class Detection:
-    classification: str
-    confidence: float
-    x: float
-    y: float
-    w: float
-    h: float
-    frame: int
+import OTAnalytics.plugin_parser.ottrk_dataformat as ottrk_format
+
+
+class Detection(BaseModel, frozen=True, allow_population_by_field_name=True):
+    """Represents a detection belonging to a `Track`.
+
+    The detection uses the xywh bounding box format.
+
+    Extends from pydantic.BaseModel.
+    IMPORTANT: Instantiation of this class is only possible by using named parameters.
+
+    Raises:
+        ValueError: confidence not in [0,1]
+        ValueError: x < 0
+        ValueError: y < 0
+        ValueError: w < 0
+        ValueError: h < 0
+        ValueError: frame < 0
+        ValueError: track_id < 0
+
+
+    Args:
+        classification (str): class of detection.
+        confidence (float): the confidence.
+        x (float): the x coordinate component of the bounding box.
+        y (float): the y coordinate component of the bounding box.
+        w (float): the width component of the bounding box.
+        h (float): the height component of the bounding box.
+        frame (int): the frame that the detection belongs to.
+        occurrence (datetime): the time of the detection's occurence.
+        input_file_path (Path): absolute path to otdet that the detection belongs to
+        at the time of its creation.
+        interpolated_detection (bool): whether this detection is interpolated.
+        track_id (int): the track id this detection belongs to.
+    """
+
+    classification: str = Field(alias=ottrk_format.CLASS)
+    confidence: float = Field(gt=0, le=1, description="must be between range [0,1]")
```

Comment:
Range `(0,1]` which would be: `gt=0` and `le=1` or range `[0,1]` which would be `ge=0` and `le=1`?

Reply:
As description I suggest: "must be in range"

Reply:
Thanks, that's correct! I will fix that!